app/models/user.rb

```diff
@@ -47,7 +47,28 @@ def reset_omniauth
     self.provider = nil
   end
 
-  def self.from_omniauth(auth)
-    where(provider: auth.provider, uid: auth.uid).first
+  class << self
+    def new_with_session(_, session)
```

Comment:
please add a comment: What is a session in this context

Reply:
This method is called by devise when a resource is built, which is done automatically by most controllers, or in this case the users_controller, with the call to `load_resource` call